### ISSUE TITLE
Dependency conflict in development requirements.

### ISSUE THREAD

**Issue by kiraware** [apragacz/django-rest-registration#211]
### Checklist

*   [x] I read [Contribution Guidelines](https://github.com/apragacz/django-rest-registration/blob/master/CONTRIBUTING.md#issues)
*   [x] I searched existing issues before opening this one
*   [x] This is not a major security issue
*   [x] I reproduced the bug with the newest version

#### Optional checklist

*   [x] https://github.com/apragacz/django-rest-registration/pull/210

### Describe the bug
Try `make install_dev`, but it complaint about dependency conflict

### Expected behavior
Should able to `make install_dev`

### Actual behavior
Fail `make install_dev`

### Steps to reproduce
1. Clone this repo
2. Create virtual environment
3. `make install_dev`

### Diagnostic info

This problem relate with this file`requirements/requirements-dev.lock.txt`

```
The conflict is caused by:
    The user requested asgiref==3.5.0
    django 4.1.7 depends on asgiref<4 and >=3.5.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
WARNING: You are using pip version 21.1.1; however, version 23.0.1 is available.
You should consider upgrading via the '/home/circleci/.venv/bin/python -m pip install --upgrade pip' command.
make: *** [Makefile:53: install_test] Error 1

Exited with code exit status 2

CircleCI received exit code 2
```

### Additional context
Nothing

**Comment by apragacz** [comment]
@kiraware thanks for sharing this. I'm aware of that issue but due to personal reasons I recently did not have enough time to finish fixing it. Will fix that one ASAP.

**Comment by apragacz** [comment]
@kiraware I unblocked the master branch and merged your changes (#210) in.
